widget/hexwidget/hexwidget.go

```diff
@@ -0,0 +1,240 @@
+		float32(h.hex.hexHeight)+theme.Padding()*2,
+	)
+}
+
+func (h *hexRenderer) Layout(size fyne.Size) {
+}
+
+func (h *hexRenderer) ApplyTheme() {
+}
+
+func (h *hexRenderer) BackgroundColor() color.Color {
+	return theme.BackgroundColor()
+}
+
+func (h *hexRenderer) Refresh() {
+	hexSegmentWidth := 0.2 * h.hex.hexWidth
+	hexSegmentVLength := (9.14 / (2 * 14)) * h.hex.hexHeight
+	hexSegmentHLength := (4.8 / 7.5) * h.hex.hexWidth
+
+	h.segmentObjects[0].(*canvas.Line).StrokeWidth = float32(hexSegmentWidth / 2)
+	h.segmentObjects[1].(*canvas.Line).StrokeWidth = float32(hexSegmentWidth / 2)
+	h.segmentObjects[2].(*canvas.Line).StrokeWidth = float32(hexSegmentWidth / 2)
+	h.segmentObjects[3].(*canvas.Line).StrokeWidth = float32(hexSegmentWidth / 2)
+	h.segmentObjects[4].(*canvas.Line).StrokeWidth = float32(hexSegmentWidth / 2)
+	h.segmentObjects[5].(*canvas.Line).StrokeWidth = float32(hexSegmentWidth / 2)
+	h.segmentObjects[6].(*canvas.Line).StrokeWidth = float32(hexSegmentWidth / 2)
+
+	pos := image.Pt(0, 0)
+
+	pt0Center := image.Pt(pos.X+int(h.hex.hexWidth/2.0+h.hex.hexOffset), pos.Y)
+	pt05 := image.Pt(int(float32(pt0Center.X)-(hexSegmentHLength/2)), pt0Center.Y)
+	pt01 := image.Pt(int(float32(pt0Center.X)+(hexSegmentHLength/2)), pt0Center.Y)
+
+	pt6Center := image.Pt(pos.X+int(h.hex.hexWidth/2.0), int(float32(pt0Center.Y)+hexSegmentVLength))
+	pt65 := image.Pt(int(float32(pt6Center.X)-(hexSegmentHLength/2)), pt6Center.Y)
+	pt61 := image.Pt(int(float32(pt6Center.X)+(hexSegmentHLength/2)), pt6Center.Y)
+
+	pt3Center := image.Pt(pos.X+int(h.hex.hexWidth/2.0-h.hex.hexOffset), int(float32(pt0Center.Y)+2*hexSegmentVLength))
+	pt34 := image.Pt(int(float32(pt3Center.X)-(hexSegmentHLength/2)), pt3Center.Y)
+	pt32 := image.Pt(int(float32(pt3Center.X)+(hexSegmentHLength/2)), pt3Center.Y)
+
+	setLineEndpoints(h.segmentObjects[0].(*canvas.Line), pt05, pt01)
+	setLineEndpoints(h.segmentObjects[1].(*canvas.Line), pt01, pt61)
+	setLineEndpoints(h.segmentObjects[2].(*canvas.Line), pt61, pt32)
+	setLineEndpoints(h.segmentObjects[3].(*canvas.Line), pt32, pt34)
+	setLineEndpoints(h.segmentObjects[4].(*canvas.Line), pt34, pt65)
+	setLineEndpoints(h.segmentObjects[5].(*canvas.Line), pt65, pt05)
+	setLineEndpoints(h.segmentObjects[6].(*canvas.Line), pt65, pt61)
+
+	for i, v := range h.segmentObjects {
+		v.(*canvas.Line).StrokeColor = h.hex.getSegmentColor(i)
+		canvas.Refresh(v)
+	}
+}
+
+func (h *hexRenderer) Destroy() {
+}
+
+func (h *hexRenderer) Objects() []fyne.CanvasObject {
+	return h.segmentObjects
+}
+
+// HexWidget represents a 7-segment hexadecimal display. The segments
+// of the display mapped active-low onto 7 state bits, with segment 0 in
+// the least significant bit.
+//
+//       0
+//     -----
+//    |     |
+//  5 |     | 1
+//    |  6  |
+//     -----
+//    |     |
+//  4 |     | 2
+//    |  3  |
+//     -----
+type HexWidget struct {
+	widget.BaseWidget
+	segments uint8
+
+	// size in "pixels" of the hex widget
```

Comment:
Not pixels, sorry.
Also, why not just use `fyne.Size` for a single field?

Reply:
2f597c1